### ISSUE TITLE
In turn-based battles, non-blocking attacks delayed by X attacks will delay up to X turns

### ISSUE THREAD

**Issue by rversteegen** [ohrrpgce/ohrrpgce#1118]
The "Delay/Advance attack by X attacks" setting (turn-based battles only) is meant to change the order in which attacks happen during a single turn, but it seems to cause a delay by that many turns instead if combined with a turn delay

> Feenick: another attack delay-related bug: it seems that attacks delayed by attacks can go off into another turn [as opposed to being guaranteed to be at the end of the turn, as I was intending to use the function]
> tmc: is the "Delay doesn't block further actions" bit on? is the attack part of a chain?
> Feenick: Yeah, both of those are the case
> tmc: is it the start of the chain? end? does the rest of the chain have any delays? I think I'd have to see the .rpg
> Feenick: It's a one-turn delay with an additional 1-attack delay. It's the second in a chain

Here's a testcase Feenick sent me; the hero's attack chains to an attack with a delay of 1 turn and 4 attacks: [delaytest.zip](https://github.com/ohrrpgce/ohrrpgce/files/4529807/delaytest.zip)
You can see the attacks build up in the queue.

Incidentally this "Delay/advance attack" setting is so confusingly named that I was initially very confused, and then really struggled to refer to it while writing this bug.

**Comment by rversteegen** [comment]
I've found that the cause of the bug is here in `pending_attacks_for_this_turn`:
```
 'Check for queued attacks
 FOR i as integer = 0 TO UBOUND(atkq)
  WITH atkq(i)
   'Ignore unused atkq() slots, attacks with a turn delay, and stunned attackers
   IF atkq_attack_active(atkq(i), bslot()) THEN
    '--only blocking queued attacks are considered part of the current
    ' turn (although it is always perfectly possible for a nonblocking
    ' attack to happen in the current turn)
    ' FIXME: Why?! This doesn't look correct - won't it cause us to the next
    ' round even if there are attacks left?
    IF .blocking THEN RETURN YES
   END IF
  END WITH
 NEXT i

 RETURN NO
```
Replacing `IF .blocking THEN RETURN YES` with `RETURN YES` fixes the problem.
So the bug doesn't happen if the attack is blocking. However there's no way I'm making that change right before release. 

I added that FIXME comment a year and a half ago but because the comment above it so confidently claimed that this was correct I didn't want to change it.

Also in r10640 (d82875bf) I added a comment about attacks (in turn-based mode) getting "shunted" to the next round even if they don't have a turn delay, and fixed another resulting bug. I'm pretty sure I was referring to this behaviour. That fix might need revisiting too.

**Comment by rversteegen** [comment]
Fixed r11813 (82a636c4ec)
The testcase Feenick provided has an attack with a delay of 1 turn and 4 attacks, but the bug happens if the turn delay is zero too. Therefore I think lots of games encounter/rely on this bug, so I've added a "Non-turn attack delays can also cause turn delays" backcompat bit to get the old behaviour.
